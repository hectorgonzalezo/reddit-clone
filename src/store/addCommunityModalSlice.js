import { createSlice } from '@reduxjs/toolkit';

export const addCommunityModalSlice = createSlice({
  name: 'addCommunityModalVisible',
  initialState: false,
  reducers: {
    toggleAddCommunityModal: (state, action) => {
      return !state
    }
  },
});

export const { toggleAddCommunityModal } = addCommunityModalSlice.actions;

export const selectAddCommunityModalVisibility = (state) => state.addCommunityModalVisible;

export default addCommunityModalSlice.reducer;
