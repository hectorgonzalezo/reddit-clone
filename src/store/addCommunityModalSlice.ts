import { createSlice } from '@reduxjs/toolkit';

export const addCommunityModalSlice = createSlice({
  name: 'addCommunityModalVisible',
  initialState: false,
  reducers: {
    toggleAddCommunityModal: (state) => !state
  },
});

export const { toggleAddCommunityModal } = addCommunityModalSlice.actions;

export const selectAddCommunityModalVisibility = (state: AddCommunityModalState): boolean => state.addCommunityModalVisible;

export default addCommunityModalSlice.reducer;
