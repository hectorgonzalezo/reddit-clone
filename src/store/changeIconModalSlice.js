import { createSlice } from '@reduxjs/toolkit';

export const changeIconModalSlice = createSlice({
  name: 'changeIconModalVisible',
  initialState: false,
  reducers: {
    toggleChangeIconModal: (state, action) => {
      return !state
    }
  },
});

export const { toggleChangeIconModal } = changeIconModalSlice.actions;

export const selectChangeIconModalVisibility = (state) => state.changeIconModalVisible;

export default changeIconModalSlice.reducer;
