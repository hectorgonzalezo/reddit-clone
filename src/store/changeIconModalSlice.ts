import { createSlice } from '@reduxjs/toolkit';
import { ChangeIconModalState } from '../type';

export const changeIconModalSlice = createSlice({
  name: 'changeIconModalVisible',
  initialState: false,
  reducers: {
    toggleChangeIconModal: (state, action) => !state
  },
});

export const { toggleChangeIconModal } = changeIconModalSlice.actions;

export const selectChangeIconModalVisibility = (state: ChangeIconModalState): boolean => state.changeIconModalVisible;

export default changeIconModalSlice.reducer;
