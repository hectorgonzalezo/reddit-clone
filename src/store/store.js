import { configureStore } from '@reduxjs/toolkit';
import userReducer from './userSlice';
import subredditsReducer from './subredditsSlice';
import loginModalReducer from './loginModalSlice';
import currentSubredditReducer from './currentSubredditSlice';

const store = configureStore({
  reducer: {
    user: userReducer,
    subreddits: subredditsReducer,
    loginModalVisible: loginModalReducer,
    currentSubreddit: currentSubredditReducer,
  }
});

export default store;
