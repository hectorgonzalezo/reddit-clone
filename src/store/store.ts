import { configureStore } from '@reduxjs/toolkit';
import userReducer from './userSlice';
import postsReducer from './postsSlice';
import subredditsReducer from './subredditsSlice';
import loginModalReducer from './loginModalSlice';
import changeIconModalReducer from './changeIconModalSlice';
import addCommunityModalReducer from './addCommunityModalSlice';
import currentSubredditReducer from './currentSubredditSlice';

const store = configureStore({
  reducer: {
    user: userReducer,
    posts: postsReducer,
    subreddits: subredditsReducer,
    loginModalVisible: loginModalReducer,
    changeIconModalVisible: changeIconModalReducer,
    addCommunityModalVisible: addCommunityModalReducer,
    currentSubreddit: currentSubredditReducer,
  }
});

export default store;
