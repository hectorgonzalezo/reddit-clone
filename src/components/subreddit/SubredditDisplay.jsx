import React, { useState, useEffect } from 'react';
import { useParams } from 'react-router-dom';
import { useSelector } from 'react-redux';
import { selectSubreddits } from '../../store/subredditsSlice';
import SubredditBanner from './SubredditBanner';
import PopularPostsBar from '../PopularPostsBar';
import PostsArea from '../post/PostsArea';
import Button from '../Button';
import SubredditAbout from './SubredditAbout';
import Agreements from '../Agreements';
import '../../styles/subredditStyle.scss';

function SubredditDisplay() {
  const subredditsData = useSelector(selectSubreddits);
  const subreddit = useParams();
  const [chosenSubreddit, setChosenSubreddit] = useState(subredditsData[subreddit.name]);

  useEffect(() => {
    setChosenSubreddit(subredditsData[subreddit.name]);
  }, [subredditsData]);

  return (
    <>
      <div id="top-side">
        {chosenSubreddit !== undefined ? <SubredditBanner subreddit={chosenSubreddit} /> : null}
      </div>
      <div id="left-side">
        <PopularPostsBar />
        {chosenSubreddit !== undefined ? <PostsArea subreddits={[chosenSubreddit]} /> : null}
      </div>
      <div id="right-side">
        {chosenSubreddit !== undefined ? <SubredditAbout subreddit={chosenSubreddit} /> : null}
        <Agreements />
        <Button text="Back to Top" />
      </div>
    </>
  );
}

export default SubredditDisplay;
