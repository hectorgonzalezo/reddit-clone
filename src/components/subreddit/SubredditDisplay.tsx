import React, { useState, useEffect } from 'react';
import { useParams } from 'react-router-dom';
import { useSelector, useDispatch } from 'react-redux';
import { selectUser } from '../../store/userSlice';
import { selectSubreddits } from '../../store/subredditsSlice';
import SubredditBanner from './SubredditBanner';
import PopularPostsBar from '../PopularPostsBar';
import PostsArea from '../post/PostsArea';
import Button from '../Button';
import SubredditAbout from './SubredditAbout';
import Agreements from '../Agreements';
import CreatePostPreview from '../CreatePostPreview';
import { authorization } from '../../firebase/firebase';
import { changeCurrentSubreddit } from '../../store/currentSubredditSlice';
import '../../styles/subredditStyle.scss';

function SubredditDisplay() {
  const subredditsData = useSelector(selectSubreddits);
  const user = useSelector(selectUser);
  const { name } = useParams();
  const [chosenSubreddit, setChosenSubreddit] = useState();
  const [postsOrder, setPostsOrder] = useState('hot');
  const dispatch = useDispatch();

  function changeOrder(newOrder) {
    setPostsOrder(newOrder);
  }

  useEffect(() => {
    setChosenSubreddit(subredditsData[name]);
    dispatch(changeCurrentSubreddit(name));
    setPostsOrder(postsOrder);
  }, [subredditsData, name]);

  return (
    <>
      {chosenSubreddit !== undefined ? <SubredditBanner subreddit={chosenSubreddit} /> : null}
      <div id="left-side">
        {user.subreddits !== undefined && user.subreddits.includes(name) ? <CreatePostPreview /> : null}
        <PopularPostsBar changeOrder={changeOrder}/>
        {chosenSubreddit !== undefined ? <PostsArea subreddits={{[chosenSubreddit.name] : chosenSubreddit}} order={postsOrder} /> : null}
      </div>
      <div id="right-side">
        {chosenSubreddit !== undefined ? <SubredditAbout subreddit={chosenSubreddit} /> : null}
        <Agreements />
        <Button text="Back to Top" onClick={() => window.scrollTo({ top: 0 })}/>
      </div>
    </>
  );
}

export default SubredditDisplay;
