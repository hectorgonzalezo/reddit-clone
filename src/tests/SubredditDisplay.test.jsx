import React from 'react';
import { render, screen, container, act } from '@testing-library/react';
import { Provider } from 'react-redux';
import { BrowserRouter, MemoryRouter } from 'react-router-dom';
import '@testing-library/jest-dom';
import userEvent from '@testing-library/user-event';
import SubredditDisplay from '../components/subreddit/SubredditDisplay';
import store from '../store/store';

jest.mock('react-router-dom', () => ({
  ...jest.requireActual('react-router-dom'),
  useParams: () => ({
    name: 'aww',
  }),
}));

const mockSubreddit ={
  name: 'aww',
  dateCreated: (new Date()).toString(),
  description:
      'Things that make you go AWW! Like puppies, bunnies, babies, and so on... A place for really cute pictures and videos!',
  icon: 'https://firebasestorage.googleapis.com/v0/b/reddit-clone-83ce9.appspot.com/o/subredditIcons%2Faww_icon.jpeg?alt=media&token=2e8d3f04-c62e-44a2-86e9-daf26d307afc',
  members: 0,
  postQuantity: 3,
  subtitle: 'A subreddit for cute and cuddly pictures',
};

const { name, dateCreated, description, icon, members, postQuantity, subtitle } = mockSubreddit;

describe('Subreddit display', () => {
  test('Render subreddit info', async () => {
    await act(async () => {
      store.dispatch({
        type: 'subreddits/addSubreddit',
        payload: [mockSubreddit],
      });
    });

    // render mock post preview
    render(
      <Provider store={store}>
        <BrowserRouter>
          <SubredditDisplay />
        </BrowserRouter>
      </Provider>
    );

    expect(screen.getByText(`r/${name}`)).toBeInTheDocument();
    expect(screen.getByText(subtitle)).toBeInTheDocument();
    expect(screen.getByText(description)).toBeInTheDocument();
    expect(screen.getByTestId('members-num')).toHaveTextContent(members);
    expect(container).toMatchSnapshot();
  });
});
