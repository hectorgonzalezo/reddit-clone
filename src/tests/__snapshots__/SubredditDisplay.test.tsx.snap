// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Subreddit display Render subreddit info 1`] = `undefined`;
